torch_geometric/transforms/rooted_subgraphs.py

```diff
@@ -0,0 +1,197 @@
+import torch
+from torch_sparse import SparseTensor
+
+from torch_geometric.data import Data
+from torch_geometric.transforms import BaseTransform
+
+
+class RootedSubgraphData(Data):
+    r"""A data object describing a homogeneous graph together with each node's
+    rooted subgraph. It contains several additional properties that hold the
+    information to map to batch of every node's rooted subgraph.
+
+    Additional Properties:
+        n_id (Tensor): index of nodes includes in combined rooted subgraph.
+        e_id (Tensor): index of edges included in rcombined ooted subgraph.
+        n_sub_batch (Tensor): batch index for nodes in rooted subgraph index.
+        e_sub_batch (Tensor): batch index for edges in rooted subgraph index.
+    """
+    def __inc__(self, key, value, *args, **kwargs):
+        if key in ('n_sub_batch', 'e_sub_batch'):
+            return 1 + getattr(self, 'n_sub_batch')[-1]
```

Comment:
```suggestion
            return 1 + getattr(self, key)[-1]
```
or
```suggestion
return 1 + self.n_sub_batch[-1]
```

Reply:
Going with the second (thanks) - in both cases you ned to use `n_sub_batch` so cannot use `key`